### PR TITLE
[SPARK-36882][PYTHON] Support ILIKE API on Python

### PR DESCRIPTION
**Pull request by yoda-mon**
### What changes were proposed in this pull request?

Support ILIKE (case insensitive LIKE) API on Python.

### Why are the changes needed?

ILIKE statement on SQL interface is supported by SPARK-36674.
This PR will support Python API for it.

### Does this PR introduce _any_ user-facing change?

Yes. Users can call `ilike` from Python.

### How was this patch tested?

Unit tests.

### CODE REVIEW

File: python/pyspark/sql/column.py
Comment (HyukjinKwon) [inline python/pyspark/sql/column.py:513]: I know it is consistent with other places in the docs but let's add versionadded directive. I think we should add them actually for all places.

**Review by HyukjinKwon — APPROVED** [review]
Let's also add it https://github.com/apache/spark/blob/master/python/pyspark/sql/column.pyi#L87.

LGTM otherwise. cc @sarutak

File: python/pyspark/sql/column.py
Comment (yoda-mon) [reply python/pyspark/sql/column.py:513]: Thank you, I'm also concerned about the versionadded directive... I will add it.

**Comment by HyukjinKwon** [comment]
@yoda-mon, FYI, there's a R linter failure going up globally that seems like an issue from external server (e.g., CRAN). So feel free to ignore it for now.

**Comment by HyukjinKwon** [comment]
cc @MaxGekk too FYI

**Review by sarutak — APPROVED** [review]
LGTM, pending CIs.

**Review by MaxGekk — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
I'll leave it to @sarutak.

**Comment by sarutak** [comment]
Merged to `master`. Thank you @yoda-mon @HyukjinKwon @MaxGekk !

**Pull request closed without merging**
